src/hostapi/asio/pa_asio.cpp

```diff
@@ -3186,28 +3248,52 @@ static void sampleRateChanged(ASIOSampleRate sRate)
     // AES/EBU or S/PDIF digital input at the audio device.
     // You might have to update time/sample related conversion routines, etc.
 
-    (void) sRate; /* unused parameter */
-    PA_DEBUG( ("sampleRateChanged : %d \n", sRate));
+    double opt = sRate;
+
+    PA_DEBUG( ("asioSampleRateChanged : %f hz\n", opt));
+
+    if (!theAsioStream)
+    {
+        /* Some ASIO drivers will fire messages during OpenStream.  We ignore them. */
+        PA_DEBUG( ("    ...message blocked, device not open yet.\n"));
+        return;
+    }
+
+    fireMessageCallback( theAsioStream, paAsioSampleRateChanged, 0, 0, &opt );
 }
 
 static long asioMessages(long selector, long value, void* message, double* opt)
 {
 // TAKEN FROM THE ASIO SDK
     // currently the parameters "value", "message" and "opt" are not used.
+    int doClientCallback = 0, paAsioMessageType = 0;
     long ret = 0;
 
     (void) message; /* unused parameters */
     (void) opt;
 
     PA_DEBUG( ("asioMessages : %d , %d \n", selector, value));
 
+    if (!theAsioStream)
+    {
+        /* Some ASIO drivers will fire messages during OpenStream.  We ignore them. */
```

Comment:
This seems like incomplete code (same in the  `sampleRateChanged` handler). Is there a rationale for dropping these messages?

To me, it seems that if we're going to pass through driver messages, we should pass through all of them. All it would take would be to stash the required callback data for use here even before the stream is fully created.

Reply:
It has been quite a while since I visited this code.  My faint recollection is that some ASIO drivers were generating spurious ResetRequest messages at this time but I'd need to look into it.